### ISSUE TITLE
Package dependencies

### ISSUE THREAD

**Issue by isteves** [NCEAS/arcticdatautils#79]
I tried installing arcticdatautils on a PC, and I needed to install the following six packages before it would work:

- digest
- stringi
- xml2
- hash
- redland
- parsedate

The first three were relatively straightforward because I got this message: `ERROR: dependencies 'digest', 'stringi', 'xml2' are not available for package 'arcticdatautils'`. The other three packages showed up one at a time with this error: `Error in loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = VI[[i]]) : there is no package called 'hash'   ERROR: lazy loading failed for package 'arcticdatautils'`

I'm not sure where the last three packages show up in arcticdatautils since I can't find them in the code anywhere, but somehow they were throwing errors.

@amoeba any insight?

**Comment by amoeba** [comment]
Okay, on a fresh Windows 10 install with a fresh R install, I get this when running `remotes::install_github("nceas/arcticdatautils")`

```
** building package indices
** installing vignettes
** testing if installed package can be loaded
*** arch - i386
Error: package or namespace load failed for 'arcticdatautils' in inDL(x, as.logical(local), as.logical(now), ...):
 unable to load shared object 'C:/Users/Bryce/Documents/R/win-library/3.4/redland/libs/i386/redland.dll':
  LoadLibrary failure:  The specified module could not be found.

Error: loading failed
Execution halted
*** arch - x64
ERROR: loading failed for 'i386'
* removing 'C:/Users/Bryce/Documents/R/win-library/3.4/arcticdatautils'
In R CMD INSTALL
Warning messages:
1: running command '"C:/PROGRA~1/R/R-34~1.4/bin/x64/R" CMD INSTALL -l "C:\Users\Bryce\Documents\R\win-library\3.4" "C:/Users/Bryce/AppData/Local/Temp/Rtmpk73BY4/remotes13387c3a4916/NCEAS-arcticdatautils-4eebe7e"' had status 1 
2: In i.p(...) :
  installation of package ‘C:/Users/Bryce/AppData/Local/Temp/Rtmpk73BY4/remotes13387c3a4916/NCEAS-arcticdatautils-4eebe7e’ had non-zero exit status
```

I then went to install redland by itself and got a warning about not being able to move the unpacked package binary into place. Then I re-ran the install and it worked. Confirmed `library(redland)` works but the install of arcticdatautils fails.

This part of the above error,

```
Error: package or namespace load failed for 'arcticdatautils' in inDL(x, as.logical(local), as.logical(now), ...):
 unable to load shared object 'C:/Users/Bryce/Documents/R/win-library/3.4/redland/libs/i386/redland.dll':
  LoadLibrary failure:  The specified module could not be found.
```

is confusing because that file specifically appears to exist

```
 Directory of C:\Users\Bryce\Documents\R\win-library\3.4\redland\libs\i386

03/29/2018  04:26 PM    <DIR>          .
03/29/2018  04:26 PM    <DIR>          ..
03/29/2018  04:26 PM         2,237,454 redland.dll
03/29/2018  04:26 PM             3,670 symbols.rds
               2 File(s)      2,241,124 bytes
               2 Dir(s)   6,302,457,856 bytes free
```

The 64-bit version also exists in its appropriate-seeming location.

**Comment by amoeba** [comment]
@isteves `hash` and `parsedate` are dependencies from the `dataone` package ,not `arcticdatautils`. The `dataone` package installed just fine for me just now.

Did you ever see any errors like what I was seeing?

**Comment by isteves** [comment]
Forgot to take note of this when I installed datamgmt/etc on my PC, but I'll try it out on the next fresh laptop I get my hands on.  Thomas should be back tomorrow to give me access to some.

**Comment by isteves** [comment]
I tried uninstalling some packages (`remove.packages(c("redland", "hash", "parsedate", "dataone", "datapack", "arcticdatautils"))`) and reinstalling arcticdatautils (`devtools::install_github("NCEAS/arcticdatautils")`) to see if I got the error.  I didn't, but I might try again on a PC at some point.

I also dug a bit into package dependencies and found the [miniCRAN](https://github.com/andrie/miniCRAN) package for visualizing package relationships. Unfortunately, only for packages that are already on CRAN.  ...doesn't really solve the issue, but confirms that the package dependency issues were not really the fault of `arcticdatautils`

`plot(makeDepGraph("dataone"))`
![image](https://user-images.githubusercontent.com/25118334/38529821-1bf56222-3c1c-11e8-9660-cb3b13afa0e9.png)

`plot(makeDepGraph("datapack"))`
![image](https://user-images.githubusercontent.com/25118334/38530018-1b1b149a-3c1d-11e8-831c-c8cd2f446987.png)
